src/components/container/ProviderSearch/ProviderSearch.tsx

```diff
@@ -1,125 +1,161 @@
+    innerRef?: React.Ref<HTMLDivElement>;
 }
 
-export type ProviderSearchPreviewState = "Default" | "Searching";
-
 let currentRequestID = 0;
 
 /** Supports searching for Providers, and Health Plans for the purpose of connecting to participant data
  */
 export default function ProviderSearch(props: ProviderSearchProps) {
-    const [linkedExternalAccounts, setLinkedExternalAccounts] = useState<{ [id: number]: ExternalAccount; }>({});
+    const [featuredProviders, setFeaturedProviders] = useState<ExternalAccountProvider[] | undefined>(props.featuredProviders);
+    const [externalAccounts, setExternalAccounts] = useState<ExternalAccount[]>();
     const [searchResults, setSearchResults] = useState<ExternalAccountProvider[]>([]);
     const [searching, setSearching] = useState(true);
-    const [searchString, _setSearchString] = useState("");
+    const [searchString, setSearchString] = useState('');
     const [currentPage, setCurrentPage] = useState(0);
     const [totalResults, setTotalResults] = useState(0);
-    const addNewProviderUrl = "https://help.mydatahelps.org/hc/en-us/requests/new?ticket_form_id=34288897775635";
+    const [initialized, setInitialized] = useState<boolean>(false);
+
+    const pageSize = 45;
+
+    const operatingMode = props.operatingMode ?? ((props.publicProviderSearchApiUrl || props.publicFeaturedProvidersApiUrl) ? 'unauthenticated' : 'authenticated');
+
+    useEffect(() => {
+        setFeaturedProviders(props.featuredProviders);
+        setExternalAccounts(undefined);
+        setSearchResults([]);
+        setSearching(true);
+        setSearchString('');
+        setCurrentPage(0);
+        setTotalResults(0);
+        setInitialized(false);
+    }, [
+        props.previewState,
+        props.providerCategories, props.publicProviderSearchApiUrl,
+        props.featuredProviders, props.featuredProvidersContext, props.publicFeaturedProvidersApiUrl
+    ]);
+
+    const applyPreviewState = (previewState: ProviderSearchPreviewState): void => {
+        const previewData = createPreviewData(previewState);
+        if (!featuredProviders) {
+            setFeaturedProviders([]);
+            return;
+        }
+        if (!externalAccounts) {
+            setExternalAccounts(previewData.externalAccounts);
+            return;
+        }
+        if (previewState !== 'Searching') {
+            updateSearchResults(previewData.searchResults);
+            setTotalResults(previewData.searchResults.length);
+            setSearching(false);
+        }
+        setInitialized(true);
+    };
 
-    const searchStringRef = useRef(searchString);
-    const setSearchString = (data: string) => {
-        searchStringRef.current = data;
-        _setSearchString(data);
+    const loadFeaturedProviders = async (): Promise<void> => {
+        try {
+            const featuredProviders = operatingMode === 'authenticated'
+                ? await MyDataHelps.getFeaturedProviders(props.featuredProvidersContext)
```

Comment:
Can't remember...when no context is passed is there a default set of featured providers that comes back down?  (I think we do want there to be, or a default context, such that the default set of featured providers is present without ANY changes to props on this component etc.)

Reply:
(I assume so, otherwise we'd need to branch this to check for a featuredProvidersContext being present)

Reply:
Yes, the API returns the "Default" set of featured providers if you do not specify a `featuredProvidersContext` context or the `featuredProvidersContext` you did specify cannot be found.

Reply:
👍